connectivity/drivers/ble/FEATURE_BLE/TARGET_Ambiq_Micro/CMakelists.txt

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(mbed-os-ble-apollo3 INTERFACE)
+
+mbed_add_cmake_directory_if_labels("TARGET")
+
+target_include_directories(mbed-os-ble-apollo3
+    INTERFACE
+        hal/apollo3
+)
+
+target_sources(mbed-os-ble-apollo3
+    INTERFACE
+        hal/hci_drv_apollo3.c
+)
```

Comment:
Based on the comments above, you should have 
`target_link_libraries(mbed-os-ble-apollo3 INTERFACE <MBED_CMAKE_TARGET>)` here.